examples/emu_mps/optimatrix/helpers.py

```diff
@@ -57,6 +39,10 @@ def permute_sequence_registers(
     reg: pulser.Register, permutation: list
 ) -> pulser.Register:
     values = list(reg.qubits.values())
-    new_values = optimatrix.permute_list(values, permutation)
-    new_register = pulser.Register(dict(enumerate(new_values)))
+
+    permuted_reg = [None] * len(values)
+    for i, p in enumerate(permutation):
+        permuted_reg[i] = values[p]
+
+    new_register = pulser.Register(dict(enumerate(permuted_reg)))
```

Comment:
`new_register = pulser.Register({
    i: values[permutation[i]] for i in range(len(values))
  })`
should do the same without creating extra `permuted_reg`

Reply:
done